vm/src/stdlib/collections.rs

```diff
@@ -472,6 +472,32 @@ mod _collections {
             self.borrow_deque().len()
         }
 
+        #[pymethod(magic)]
+        fn add(&self, other: PyObjectRef, vm: &VirtualMachine) -> PyResult<Self> {
+            if other.payload_is::<PyDeque>() {
```

Comment:
You can use `payload_if_subclass` to get a handle of `other` as a `PyDeque`. Then you can clone its `VecDeque` directly and add it to the elements of `self`'s `VecDeque`.

Reply:
Thanks for the hint! I refactored it.